ldiff --git a/org/apache/commons/math/optimization/linear/SimplexSolver.java b/org/apache/commons/math/optimization/linear/SimplexSolver.java
index 60a1b3a..16d3bae 100644
--- a/org/apache/commons/math/optimization/linear/SimplexSolver.java
+++ b/org/apache/commons/math/optimization/linear/SimplexSolver.java
@@ -77,10 +77,9 @@ public class SimplexSolver extends AbstractLinearOptimizer {
         double minRatio = Double.MAX_VALUE;
         Integer minRatioPos = null;
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getHeight(); i++) {
-            final double rhs = tableau.getEntry(i, tableau.getWidth() - 1);
-            final double entry = tableau.getEntry(i, col);
-            if (MathUtils.compareTo(entry, 0, epsilon) > 0) {
-                final double ratio = rhs / entry;
+            double rhs = tableau.getEntry(i, tableau.getWidth() - 1);
+            if (MathUtils.compareTo(tableau.getEntry(i, col), 0, epsilon) >= 0) {
+                double ratio = rhs / tableau.getEntry(i, col);
                 if (ratio < minRatio) {
                     minRatio = ratio;
                     minRatioPos = i; 
